src/crimson/os/seastore/journal/circular_bounded_journal.h

```diff
@@ -30,7 +30,27 @@ using NVMeBlockDevice = nvme_device::NVMeBlockDevice;
 /**
  * CircularBoundedJournal
  *
- * TODO: move record from CircularBoundedJournal to RandomBlockManager
+ * 
+ * CircularBoundedJournal (CBJournal) is the journal that works like circular
+ * queue. With CBJournal, Seastore will append some of the records if the size
+ * of the record is small (most likely metadata), at which point the head
+ * (written_to) will be moved. Then, eventually, Seastore applies the records
+ * in CBjournal to RBM---the tail (applied_to) is advanced (TODO).
+ *
+ * - Commit time
+ * After submit_record is done, last_committed_record_base and written_to are
+ * advanced---last_committed_record_base indicates the last committed record and
+ * written_to represents where the new record will be appended. Note that
+ * applied_to is not changed here.
+ *
+ * - Replay time
+ * At replay time, CBJournal begins to replay records in CBjournal by reading
+ * records from applied_to. Then, CBJournal examines whether the records is valid
```

Comment:
Is applied_to a header field?  When is the ondisk field updated?

Reply:
> Is applied_to a header field? When is the ondisk field updated?

Yes. 
After the apply work---applying the records in CBJournal to RBM---is done.

Reply:
I don't think this is actually correct.  Applying records to RBM means that *those* deltas won't need to be replayed, but the CBJournal will in general include deltas for records on SegmentedDevices as well.  I don't think applied_to is actually relevant to the replay semantics of CBJournal, it simply restricts which RBM write-ahead deltas can be ignored.  Even then, I think its only an optimization.

Reply:
I didn't consider that CBjournal can include records on SegmentedDevices. I added journal_tail when replaying records. See the last updated commit.